controllers/datadoghq/recommender.go

```diff
@@ -58,8 +92,31 @@ type ReplicaRecommendationResponse struct {
 	Details            string
 }
 
-func NewRecommenderClient() RecommenderClient {
-	return &RecommenderClientImpl{}
+func NewRecommenderClient(client *http.Client) RecommenderClient {
+	return &RecommenderClientImpl{
+		client: client,
+	}
+}
+
+func (r *RecommenderClientImpl) instrumentedClient(recommender string) *http.Client {
+	client := *r.client
```

Comment:
This code _can_ be misleading, it's definitely possible to change behaviour of original `client` still. In that case `.Transport` is only changed and you can hope that the wrapping code does create a new RT and not modify the existing one but there's no guarantee on that.

I think it warrants at least a comment.

Reply:
done